src/connectionhealth.cpp

```diff
@@ -275,3 +279,84 @@ void ConnectionHealth::overwriteStabilityForInspector(
   m_stability = stability;
   emit stabilityChanged();
 }
+
+void ConnectionHealth::startTimingDistributionMetric(
+    ConnectionStability stability) {
+#if defined(MZ_WINDOWS) || defined(MZ_LINUX) || defined(MZ_MACOS) || \
+    defined(MZ_DUMMY)
+  switch (stability) {
+    case ConnectionHealth::Unstable:
+      m_metricsTimerId =
+          mozilla::glean::connection_health::unstable_time.start();
+      break;
+    case ConnectionHealth::NoSignal:
+      m_metricsTimerId =
+          mozilla::glean::connection_health::no_signal_time.start();
+      break;
+    default:
+      m_metricsTimerId = mozilla::glean::connection_health::stable_time.start();
+  }
+#endif
+}
+
+void ConnectionHealth::stopTimingDistributionMetric(
+    ConnectionStability stability) {
+#if defined(MZ_WINDOWS) || defined(MZ_LINUX) || defined(MZ_MACOS) || \
+    defined(MZ_DUMMY)
```

Comment:
nit: Maybe we can define a macro like "MZ_MOBILE" instead that is true when we are in IOS and ANDROID instead of this?

Reply:
Done, thanks.